src/btree/node.rs

```diff
@@ -561,6 +561,32 @@ impl Node {
 		Ok(())
 	}
 
+	pub fn seek_to_last(
+		from: Self,
+		values: TablesRef,
+		log: &impl LogQuery,
+		depth: u32,
+		stack: &mut Vec<(usize, NodeType, Self)>,
+	) -> Result<()> {
+		let i = if let Some(i) = from.last_child_index().or_else(|| from.last_separator_index()) {
+			i
+		} else {
+			return Ok(())
+		};
+
+		if depth != 0 {
+			if let Some(child) = from.fetch_child(i, values, log)? {
+				if i > 0 {
+					stack.push((i - 1, NodeType::Separator, from));
+				}
+				return Self::seek_to_last(child, values, log, depth - 1, stack)
```

Comment:
Hmm, I believe it will also work without recursing but it will be lazy. I can change that if you insist, it will make code simpler I think :man_shrugging: